### PR TITLE
Remove unused components from `ui/`

### PR DESCRIPTION
**Pull request by brookewp**
## What?

This expands on the work started in #52953 to remove unused components from the `packages/components/src/ui` folder. 

## Why?

These components (`ui/tooltip` and `ui/shortcut`) are not publicly exported and are relatively unused. There was one internal use of `ui/tooltip`, which is replaced in this PR. 

## How?

By replacing the only internal usage of `ui/tooltip` with our Tooltip component and deleting the unused components. 

## Testing Instructions

`npm run build` without errors 

### Colorpicker: 

1. `npm run storybook:dev` 
2. Navigate to the `ColorPicker` component 
3. Ensure the tooltip for the copy button appears* and behaves the same as trunk

*The placement will look off in Storybook unless the sidebar is hidden (key-down `S` to hide) or the story is opened in isolation. The tooltip background is also black instead of grey, but that reflects a design change made earlier: #50792

### CODE REVIEW

File: packages/components/src/ui/tooltip/types.ts [hunk 1]
```diff
@@ -1,67 +0,0 @@
...
- * External dependencies
- */
-// eslint-disable-next-line no-restricted-imports
-import type { TooltipInitialState, TooltipProps } from 'reakit';
-import type { ReactElement, ReactNode } from 'react';
-
-/**
- * Internal dependencies
- */
-import type { PopperProps } from '../utils/types';
```
Comment (tyxla) [inline packages/components/src/ui/tooltip/types.ts:11]: I think that the instance of `PopperProps` along with its `PopperPlacement` type and the `SizeRangeDefault` type can be removed from `/ui/utils/types`.

File: packages/components/src/ui/tooltip/styles.js [hunk 2]
```diff
@@ -1,45 +0,0 @@
...
- * External dependencies
- */
-import { css } from '@emotion/react';
-import styled from '@emotion/styled';
-
-/**
- * Internal dependencies
- */
-import { Shortcut } from '../shortcut';
-import * as ZIndex from '../../utils/z-index';
```
Comment (tyxla) [inline packages/components/src/ui/tooltip/styles.js:11]: I think this might be the last usage of the `z-index` utils, so it could be removed.

File: packages/components/src/tooltip/index.tsx [hunk 3]
```diff
@@ -73,7 +73,6 @@ function Tooltip( props: TooltipProps ) {
 	return (
 		<>
 			<Ariakit.TooltipAnchor
-				onBlur={ tooltipStore.hide }
```
Comment (tyxla) [inline packages/components/src/tooltip/index.tsx:76]: Could you please elaborate on this change?

**Review by tyxla — APPROVED** [review]
The copy button in the color picker tests well for me. ✅ 

There might be a few potential remnants to clean up, but this looks great anyway!  🚀 

Nice to see a 4% reduction of the components package bundle size 💯

File: packages/components/src/color-picker/color-copy-button.tsx [hunk 4]
```diff
@@ -56,20 +55,17 @@ export const ColorCopyButton = ( props: ColorCopyButtonProps ) => {
...
+				copiedColor === color.toHex() ? __( 'Copied!' ) : __( 'Copy' )
 			}
-			placement="bottom"
 		>
 			<CopyButton
-				isSmall
 				ref={ copyRef }
 				icon={ copy }
 				showTooltip={ false }
+				size="small"
```
Comment (ciampo) [inline packages/components/src/color-picker/color-copy-button.tsx]: Is this change necessary and/or related to the tooltip swap?

File: packages/components/src/ui/shortcut/component.tsx
Comment (ciampo) [inline packages/components/src/ui/shortcut/component.tsx:1]: Before deleting this component, should we assess if there's anything from its implementation that we can move to the current `Shortcut` component? It looks like the props are similar, but I'd like to make sure we're not missing out on anything before deleting the files — this could be a little nice follow-up

File: packages/components/src/tooltip/index.tsx (hunk 3 above)
Comment (brookewp) [reply packages/components/src/tooltip/index.tsx:76]: Yes, thank you for asking! I meant to leave a comment here when I opened the PR.

There was a conflict between the `onBlur` added to the Tooltip anchor and the function `useCopyToClipboard` in the `ColorPicker` component: https://github.com/WordPress/gutenberg/blob/70f787b79468c938ce2d0335ffa768e59e455cc3/packages/components/src/color-picker/color-copy-button.tsx#L21

In this function, the `focus` is reset, resulting in the tooltip hiding on copy button click before displaying the 'copied' text:

https://github.com/WordPress/gutenberg/blob/23bb9301902bb5825480240bd5aaec9d3cc3c380/packages/compose/src/hooks/use-copy-to-clipboard/index.js#L52-L59

Originally, we added the `onBlur` to the tooltip to replicate the legacy behavior where the tooltip is hidden when leaving the document. This was discovered through failing tests; but now, the tests pass without it. The [utility function](https://github.com/WordPress/gutenberg/pull/48440/files#diff-d040fe8f8aee3cd48e8f849d4fb7f848570f7f27fac6c85557a56538aa84a566) was added after, so it's likely the failure was related to leaky tests instead. 

I discussed removing this or adding it as a prop with @ciampo, and we decided to remove it for now. And if we find a use-case where it makes sense to have it, we could add it as an optional prop.

File: packages/components/src/color-picker/color-copy-button.tsx (hunk 4 above)
Comment (brookewp) [reply packages/components/src/color-picker/color-copy-button.tsx]: This isn't related to tooltip, so I can remove it. I saw the deprecation warning and thought I could update it while I was already there.

File: packages/components/src/ui/tooltip/types.ts (hunk 1 above)
Comment (brookewp) [reply packages/components/src/ui/tooltip/types.ts:11]: Good catch!

File: packages/components/src/ui/tooltip/styles.js (hunk 2 above)
Comment (brookewp) [reply packages/components/src/ui/tooltip/styles.js:11]: Another good catch. 🙂

File: packages/components/src/ui/shortcut/component.tsx
Comment (brookewp) [reply packages/components/src/ui/shortcut/component.tsx:1]: Are you suggesting looking at it in a follow-up and separating it from this PR? To me, it doesn't appear there is anything to add, as they appear nearly identical. But I'm happy to look over it with you, in case I'm missing something.

Comment (ciampo) [reply packages/components/src/ui/shortcut/component.tsx:1]: Out of curiosity, what is the difference?

If they are nearly identical, then feel free to delete the component (as this PR is already doing) and ignore my previous message.

Comment (brookewp) [reply packages/components/src/ui/shortcut/component.tsx:1]: The main difference is `Shortcut` returns a `span` tag whereas `ui/shortcut` returns a `View` component, but it's rendered `as` a `span`. So `ui/shortcut` has a `forwardedRef` and is connected with `contextConnect` and `useContextSystem`. 

`Shortcut` has a `types.js` file versus types added inline in `ui/shortcut`, but the logic between them is the same: 

https://github.com/WordPress/gutenberg/blob/1873738159b6ffa9bd59f034dc8254884c5e3fb6/packages/components/src/shortcut/index.tsx#L22-L36

https://github.com/WordPress/gutenberg/blob/1873738159b6ffa9bd59f034dc8254884c5e3fb6/packages/components/src/ui/shortcut/component.tsx#L33-L45

They also expose identical props. The tests are also nearly identical, with `Shortcut` having one more test to check the `className`.

File: packages/components/src/color-picker/color-copy-button.tsx (hunk 4 above)
Comment (ciampo) [reply packages/components/src/color-picker/color-copy-button.tsx]: Ah, I had missed the `isSmall` prop. Let's leave it as-is for this PR, and then we can open a PR to refactor all instances of `isSmall` to `size="small"` ?

File: packages/components/src/ui/shortcut/component.tsx
Comment (ciampo) [reply packages/components/src/ui/shortcut/component.tsx:1]: Thank you!

**Review by ciampo — APPROVED** [review]
LGTM 🚀 

Always great to see a largely negative line diff 🎉

File: packages/components/src/tooltip/index.tsx (hunk 3 above)
Comment (tyxla) [reply packages/components/src/tooltip/index.tsx:76]: Makes sense to me, thanks 👍

**Review by tyxla — APPROVED** [review]
Will obviously need a rebase before shipping, but it looks great 🚀 

It's always a pleasure to see some red!

**Review by chad1008 — APPROVED** [review]
LGTM as well! :rocket: :ship:

File: packages/components/src/tooltip/index.tsx (hunk 3 above)
Comment (ciampo) [reply packages/components/src/tooltip/index.tsx:76]: We may need to consider adding a prop to customize this behaviour, since this change is related to a regression #57206
